operator/exporter.py

```diff
@@ -0,0 +1,180 @@
+import json
+import urllib3
+import uvicorn
+import logging
+from utils import execute, CommandError
+from fastapi import FastAPI
+from prometheus_client import make_asgi_app, start_http_server, Gauge
+from kadalulib import logf, logging_setup
+
+# Define all metrics here
+total_capacity_bytes = Gauge('kadalu_storage_total_capacity_bytes', 'Kadalu Total Storage Capacity', ['name'])
+capacity_used_bytes = Gauge('kadalu_storage_used_capacity_bytes', 'Kadalu Total Storage Used Capacity', ['name'])
+capacity_free_bytes = Gauge('kadalu_storage_free_capacity_bytes', 'Kadalu Total Storage Free Capacity', ['name'])
+
+total_inodes = Gauge('kadalu_storage_total_inodes', 'Kadalu Total Storage Inodes', ['name'])
+used_inodes = Gauge('kadalu_storage_used_inodes', 'Kadalu Total Storage Inodes Used', ['name'])
+free_inodes = Gauge('kadalu_storage_free_inodes', 'Kadalu Total Storage Inodes Free', ['name'])
+
+total_pvc_capacity_bytes = Gauge('kadalu_pvc_total_capacity_bytes', 'Kadalu Total PVC Capacity', ['name'])
+capacity_pvc_used_bytes = Gauge('kadalu_pvc_used_capacity_bytes', 'Kadalu Total PVC Used Capacity', ['name'])
+capacity_pvc_free_bytes = Gauge('kadalu_pvc_free_capacity_bytes', 'Kadalu Total PVC Free Capacity', ['name'])
+
+total_pvc_inodes = Gauge('kadalu_pvc_total_inodes', 'Kadalu Total Total PVC Inodes', ['name'])
+used_pvc_inodes = Gauge('kadalu_pvc_used_inodes', 'Kadalu Total Used PVC Inodes', ['name'])
+free_pvc_inodes = Gauge('kadalu_pvc_free_inodes', 'Kadalu Total Free PVC Inodes', ['name'])
+
+memory_usage = Gauge('kadalu_memory_usage_in_bytes', 'Kadalu Memory Usage in Bytes', ['name'])
+cpu_usage = Gauge('kadalu_cpu_usage_in_ns', 'Kadalu Memory Usage in Nanoseconds', ['name'])
+
+http = urllib3.PoolManager()
+app = FastAPI()
+
+def get_pod_ip_data():
+    """ Get IP Addresses of all Pods in kadalu namespace """
+
+    cmd = ["kubectl", "get", "pods", "-nkadalu", "-ojson"]
+
+    try:
+        resp = execute(cmd)
+        data = json.loads(resp.stdout)
+        pod_ip_data = {}
+
+        for item in data["items"]:
+
+            # Skip pods which are not "Running"
+            if item["status"]["phase"] == "Pending":
+                continue
+
+            pod_name = item["metadata"]["name"]
+            ip_addr = item["status"]["podIP"]
+            pod_ip_data[pod_name] = ip_addr
```

Comment:
Change this to `pod_ip_data[pod_name] = {"ip": ip_addr}` so that you can add other details to the same dict. For example `"status": item["status"]["phase"]`

```
pod_ip_data[pod_name] = {
    "ip": ip_addr,
    "status": item["status"]["phase"]
}
```

Reply:
Also capture number of ready containers and total containers.